diff -pruN ffmpeg-3.3.2/libavformat/oggdec.c ffmpeg-3.3.2.2/libavformat/oggdec.c
--- ffmpeg-3.3.2/libavformat/oggdec.c	2017-06-07 09:03:55.000000000 +0700
+++ ffmpeg-3.3.2.2/libavformat/oggdec.c	2017-07-11 21:47:22.325439745 +0700
@@ -480,7 +480,7 @@ static int ogg_packet(AVFormatContext *s
     int complete = 0;
     int segp     = 0, psize = 0;
 
-    av_log(s, AV_LOG_TRACE, "ogg_packet: curidx=%i\n", ogg->curidx);
+//    av_log(s, AV_LOG_TRACE, "ogg_packet: curidx=%i\n", ogg->curidx);
     if (sid)
         *sid = -1;
 
@@ -495,8 +495,8 @@ static int ogg_packet(AVFormatContext *s
 
         os = ogg->streams + idx;
 
-        av_log(s, AV_LOG_TRACE, "ogg_packet: idx=%d pstart=%d psize=%d segp=%d nsegs=%d\n",
-                idx, os->pstart, os->psize, os->segp, os->nsegs);
+//        av_log(s, AV_LOG_TRACE, "ogg_packet: idx=%d pstart=%d psize=%d segp=%d nsegs=%d\n",
+  //              idx, os->pstart, os->psize, os->segp, os->nsegs);
 
         if (!os->codec) {
             if (os->header < 0) {
